src/server/project.ts

```diff
@@ -364,7 +369,7 @@ namespace ts.server {
             return addRange(result, this.typingFiles) || ts.emptyArray;
         }
 
-        private getOrCreateScriptInfoAndAttachToProject(fileName: string) {
+        private getOrCreateScriptInfoAndAttachToProject(fileName: string, isAuxiliaryFile?: boolean) {

```

Comment:
Did you think about creating auxillary project and calling methods on that project (new project kind that probably derives from inferred project)  and handling that.. THat might be simple than having to track two types of contianingProjects etc.. Also better to investigate where what is attached.

Reply:
I thought a Project subclass is probably a lot heavier of an object than I need, but I agree the ScriptInfo changes (and Project/LanguageServiceHost API changes that work with ScriptInfo) is the worst part of this PR, so maybe that would simplify things a lot.

Reply:
Project shouldnt be that heavier than having a program. (ignore all watching in this new kind of project and other extra stuff and you should have blackbox around the program thats not that heavy) I am inclined for this because project has already handled all sorts of reclaimation and update of project/scriptinfo../ Having another kind adds to complexity and issues with correct handling of attaching/detaching and removing program etc.

Reply:
The main part that looks unwieldy to me is that the `Project` base class unconditionally creates a LanguageService, which should never be used. The compiler options for the program to keep it lightweight (`noLib: true`, `diagnostics: false`) mean we don’t actually want to use this LanguageService for anything. (We could never return this kind of project from `scriptInfo.getDefaultProject`; we’d still need to create an InferredProject if an auto-importable script inside node_modules was opened by the client.)

Reply:
But even if you create LS , it will jsut create program in there since you wont use this for anything else.. If fact there is getLanguageService method on project which you can override and throw error so we know its used in unexpected ways... You can get the program from `project.getIntentionalLS().getProgram()` .. The inferred project thing shoul;d work if you accomodate exclusion of the new kind of project in all places where it shouldnt.. 
The thing i like about it is the script info/project/LS management is already sorted in these scenario and would be easy to diagnose since we already have logs in the server for changes to program and updates so just looking at the log you should be able to track if project was updated/ what files were added and so on